### PR TITLE
Merge CI fixes from main to v2.

### PR DESCRIPTION
**Pull request by shimkiv**
- https://github.com/o1-labs/o1js/pull/1760
- https://github.com/o1-labs/o1js/pull/1772

### CODE REVIEW

**Review by ymekuria — APPROVED** [review]

**Comment by mitschabaude** [comment]
Is this a proper merge? I don't see any commits

**Comment by shimkiv** [comment]
> Is this a proper merge? I don't see any commits

Yea, CI is happy now. I picked up changed files instead of concrete commits this time.

**Comment by mitschabaude** [comment]
> > Is this a proper merge? I don't see any commits
> 
> Yea, CI is happy now. I picked up changed files instead of concrete commits this time.

In general it would be better to do merges because whoever does the merge main -> v2 next time might have to deal with fake "conflicts" introduced by this PR which is making the same changes as on main but in different commits

File: .github/workflows/build-action.yml [hunk 1]
```diff
@@ -8,7 +8,53 @@ on:
   workflow_dispatch: {}
 
 jobs:
+  Prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      test_count: ${{ steps.count_tests.outputs.test_count }}
+      chunk_count: 8 # This is hardcoded to 8, but it can be changed to any number.
```
Comment (Trivo25) [inline .github/workflows/build-action.yml:15]: what does this number do?

**Comment by shimkiv** [comment]
> In general it would be better to do merges because whoever does the merge main -> v2 next time might have to deal with fake "conflicts" introduced by this PR which is making the same changes as on main but in different commits

Ok, since this PR already has different commit then I'll need to close this one and create one with only commits from `main`?

**Comment by mitschabaude** [comment]
> > In general it would be better to do merges because whoever does the merge main -> v2 next time might have to deal with fake "conflicts" introduced by this PR which is making the same changes as on main but in different commits
> 
> Ok, since this PR already has different commit then I'll need to close this one and create one with only commits from `main`?

It's fine! Just wanted to bias you towards using `git merge` next time :D

File: .github/workflows/build-action.yml (hunk 1 above)
Comment (shimkiv) [reply .github/workflows/build-action.yml:15]: Amount of chunks unit tests split into I believe.

**Comment by shimkiv** [comment]
> It's fine! Just wanted to nudge you towards using `git merge` next time :D

Thanks, yet I'll do it this time too. v2 -> main already has a good bunch of conflicts, and I don't want to cause more pain :)

**Comment by shimkiv** [comment]
https://github.com/o1-labs/o1js/pull/1781

**Pull request closed without merging**